specification/keyvault/Security.KeyVault.Secrets/models.tsp

```diff
@@ -0,0 +1,433 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+
+using Azure.Core;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+/**
+ * Reflects the deletion recovery level currently in effect for secrets in the
+ * current vault. If it contains 'Purgeable', the secret can be permanently
+ * deleted by a privileged user; otherwise, only the system can purge the secret,
+ * at the end of the retention interval.
+ */
+union DeletionRecoveryLevel {
+  string,
+
+  /**
+   * Denotes a vault state in which deletion is an irreversible operation, without
+   * the possibility for recovery. This level corresponds to no protection being
+   * available against a Delete operation; the data is irretrievably lost upon
+   * accepting a Delete operation at the entity level or higher (vault, resource
+   * group, subscription etc.)
+   */
+  Purgeable: "Purgeable",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable, and which also permits
+   * immediate and permanent deletion (i.e. purge). This level guarantees the
+   * recoverability of the deleted entity during the retention interval (90 days),
+   * unless a Purge operation is requested, or the subscription is cancelled. System
+   * wil permanently delete it after 90 days, if not recovered
+   */
+  Recoverable_Purgeable: "Recoverable+Purgeable",
```

Comment:
Is it expected that `_` is used here? If you notice in JS we use CapitalCase: https://github.com/Azure/azure-sdk-for-js/blob/c53b5a0403410fad1f65180cdbf8d5044a61970d/sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md?plain=1#L70-L78

And I'd like to understand whether this needs to be / can be done in client.tsp or here.

Reply:
I elected to use underscores in place of `+` for these so that we wouldn't need to use backticks (\`), since `+` isn't normally allowed in names. Since TSP is introducing a new `name` field into [the enum's Swagger definition](https://github.com/Azure/azure-rest-api-specs/blob/mccoyp/kv-secrets-tsp/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.5/secrets.json#L757-L791) that didn't used to be specified, I didn't have a precedent to match here.

Python does have [underscores in enum value names](https://github.com/Azure/azure-sdk-for-python/blob/fc82782cb3b22d29f9068717e7bc34d47f73fc38/sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_generated/models/_enums.py#L13), but with underscores before each capitalization instead of just replacing `+` -- I wasn't trying to make it language-specific. Python's auto-underscore approach does make me think that using CapitalCase in the spec could possibly fix JS's problem without changing what Python generates.

Would a change to these value names change break JS, though? My understanding was that no one was exposing this enum apart from in typing hints.

Reply:
We do in JS: https://github.com/Azure/azure-sdk-for-js/blob/ae05a39d0f9e313e82fc93ed37fa5c3c09e17db0/sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md?plain=1#L70-L78 but it's possibly just a rare occurrence of `+` in an enum value that I can work around (unless you're amenable to using CapitalCase 😍 )

Reply:
I just updated the enum to use CapitalCase -- let me know if that works! I confirmed locally that Python is good either way 🙂

Reply:
Im going to do the same thing for keys, just FYI - let me know if that's an issue for you!

Reply:
I assume we should also update Certificates similarly?

Reply:
Yes, if and when I finally get to it (unless you get there first) 😄